pkg/util/stringreplace/object.go

```diff
@@ -73,14 +95,27 @@ func visitUnsettableValues(typeOf reflect.Type, original reflect.Value, visitor
 	}
 	switch existing.Kind() {
 	case reflect.String:
-		s := visitor(existing.String())
-		val.Set(reflect.ValueOf(s))
+		s, asString := visitor(existing.String())
+
+		var b []byte
+		if asString {
+			b = []byte(fmt.Sprintf("{\"k\":%q}", s))
+		} else {
+			b = []byte(fmt.Sprintf("{\"k\":%s}", s))
+		}
+		var data map[string]interface{}
```

Comment:
without "asString" how does the encoder differentiate between "true" (the string) and *true* (the boolean value)?